graph/run-dynamic.go

```diff
@@ -187,7 +187,7 @@ func ConvergeDynamicThread[EP EPP[E], V VPI[V], E EPI[E], M MVI[M], N any, A Alg
 		}
 
 		// The main check for updates to topology. This occurs with priority over algorithmic messages.
-		if !strucClosed && !blockTop {
+		if !epoch && !strucClosed && !blockTop {
```

Comment:
I think we can also skip edge events when waiting for a query to finish? This if branch is very slow when there's no edge events due to backoff.